tools/istio-docker.mk

```diff
@@ -92,7 +92,7 @@ docker.proxy: pilot/docker/envoy_telemetry.yaml.tmpl
 	cp pilot/docker/*.yaml.tmpl $(DOCKER_BUILD_TOP)/proxy/
 	cp ${ISTIO_ENVOY_RELEASE_PATH} $(DOCKER_BUILD_TOP)/proxy/envoy
 	time (cd $(DOCKER_BUILD_TOP)/proxy && \
-		docker build -t $(HUB)/proxy:$(TAG) -f ${DOCKER_PROXY_CFG} .)
+		sudo docker build -t $(HUB)/proxy:$(TAG) -f ${DOCKER_PROXY_CFG} .)
```

Comment:
Why ? It'll also break stuff for lots of people who don't want to type the password on make.
Maybe add an env.

${DOCKER_SUDO}  - and set it in your env.

Reply:
Hmm, didn't notice this sneaks in. I used it for my local build, and not to be part of the PR. Let me remove it